sdk/nodejs/s3/routingRules.ts

```diff
@@ -0,0 +1,182 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * An array of `RoutingRule` objects.
+ *
+ * For more details, please refer to the AWS documentation online:
+ * https://docs.aws.amazon.com/AmazonS3/latest/dev/how-to-page-redirect.html#advanced-conditional-redirects
+ */
+export type RoutingRules = RoutingRule[];
+
+/**
+ * A rule that identifies a condition and the redirect that is applied when the condition is met.
+ * If a condition is not included, the rule is applied to all requests.
+ */
+export interface RoutingRule {
+    /**
+     * A condition that must be met for the specified redirect to be applied.
+     * If not included, the rule is applied to all requests.
+     */
+    Condition?: Condition;
+
+    /**
+     * Provides instructions for redirecting the request. You can redirect requests to another host or another page, or you can specify another protocol to use.
+     */
+    Redirect: Redirect;
+}
+
+// Multiple types are used for `Condition` to enforce at least one property being set.
+
+/**
+ * A condition that must be met for the specified redirect to be applied.
+ */
+export type Condition = HttpErrorCodeReturnedEqualsCondition | KeyPrefixEqualsCondition;
+
+export interface HttpErrorCodeReturnedEqualsCondition {
+    /**
+     * The HTTP error code that must match for the redirect to apply. If an error occurs, and if the error code meets this value, then the specified redirect applies.
+     */
+    HttpErrorCodeReturnedEquals: string;
+
+    /**
+     * The prefix of the object key name from which requests are redirected.
+     *
+     * If specified, both `HttpErrorCodeReturnedEquals` and `KeyPrefixEquals` must be true for the condition to be met.
+     */
+    KeyPrefixEquals?: string;
+}
+
+export interface KeyPrefixEqualsCondition {
+    /**
+     * The HTTP error code that must match for the redirect to apply. If an error occurs, and if the error code meets this value, then the specified redirect applies.
+     *
+     * If specified, both `HttpErrorCodeReturnedEquals` and `KeyPrefixEquals` must be true for the condition to be met.
```

Comment:
i'm confused what you mean by 'true' here since these are strings.

Reply:
These are routing conditions. The conditions are defined by strings, and depending on the route to the bucket, they'll either be true or false.

This text comes directly from the AWS docs at https://docs.aws.amazon.com/AmazonS3/latest/dev/how-to-page-redirect.html#advanced-conditional-redirects (scroll down a bit).